osquery/main/shell.cpp

```diff
@@ -83,6 +88,42 @@ int profile(int argc, char *argv[]) {
   return 0;
 }
 
+// readline completion expects strings to be malloced. readline will free them
+// later.
+char *copy_string(const std::string &str) {
+  char *copy = NULL;
+  if ((copy = (char *)malloc(str.size() + 1)) == NULL) {
```

Comment:
Can we simplify this and move the mallocing/assignment outside of the conditional? Then do a check against `nullptr` and `copy` within the conditional?